superset/db_engine_specs/base.py

```diff
@@ -640,7 +640,7 @@ def select_star(  # pylint: disable=too-many-arguments,too-many-locals
         if schema:
             full_table_name = quote(schema) + "." + quote(table_name)
         else:
-            full_table_name = quote(table_name)
+            full_table_name = ".".join([quote(t) for t in table_name.split(".")])
```

Comment:
Are we sure we want to split with `.` here? Periods are routinely used in table names in some analytical dbs (e.g. `table.csv` when the underlying data is in a csv file, for instance), and I believe this method usually gets called only after schema and table names have been properly separated. Unless this is needed for this feature, I believe we should stick to the existing logic. If not, this should be covered by a unit test that ensures that this doesn't cause a regression for table names with periods.